src/Packages/RetroMole.Render.Veldrid/Controller.cs

```diff
@@ -537,114 +524,108 @@ public void CreateDeviceResources()
                 PrimitiveTopology.TriangleList,
                 new ShaderSetDescription(vertexLayouts, new[] { _vertexShader, _fragmentShader }),
-                new ResourceLayout[] { _layout, _textureLayout },
+                new ResourceLayout[] { _mainRL, _textRL },
                 outputDescription,
                 ResourceBindingModel.Default);
             _pipeline = factory.CreateGraphicsPipeline(ref pd);
 
-            _mainResourceSet = factory.CreateResourceSet(new ResourceSetDescription(_layout,
+            _mainRS = factory.CreateResourceSet(new(
+                _mainRL,
                 _projMatrixBuffer,
-                _gd.PointSampler));
-
-            _fontTextureResourceSet = factory.CreateResourceSet(new ResourceSetDescription(_textureLayout, _fontTextureView));
+                _gd.PointSampler
+            ));
+            _ftRS = factory.CreateResourceSet(new(
+                _textRL,
+                (((Texture, TextureView))FontTexture.Texture).Item2
+            ));
         }
 
-        public override IntPtr GetOrCreateImgGuiBinding(Core.Interfaces.Texture Texture)
+        public override IntPtr BindTexture(Core.Interfaces.Texture Texture)
         {
-            var factory = _gd.ResourceFactory;
-            var t = factory.CreateTexture(new TextureDescription(
+            Texture t = _gd.ResourceFactory.CreateTexture(new TextureDescription(
                 (uint)Texture.Width,
                 (uint)Texture.Height,
                 1,1,1,
                 PixelFormat.B8_G8_R8_A8_UNorm,
                 TextureUsage.Sampled,
                 TextureType.Texture2D
-            ));
+            ));    
             
-            _gd.UpdateTexture<byte>(t, Texture.Data, 0, 0, 0, (uint)Texture.Width, (uint)Texture.Height, 1, 0, 0);
+            _gd.UpdateTexture(t, Texture.Pixels, 0, 0, 0, (uint)Texture.Width, (uint)Texture.Height, 1, 0, 0);
 
-            if (!_autoViewsByTexture.TryGetValue(t, out TextureView textureView))
-            {
-                textureView = factory.CreateTextureView(t);
-                _autoViewsByTexture.Add(t, textureView);
-                _ownedResources.Add(textureView);
-            }
+            var tv = _gd.ResourceFactory.CreateTextureView(t);
+            var rs = _gd.ResourceFactory.CreateResourceSet(new(
+                _textRL,
+                tv
+            ));
 
-            if (!_setsByView.TryGetValue(textureView, out ResourceSetInfo rsi))
-            {
-                ResourceSet resourceSet = factory.CreateResourceSet(new ResourceSetDescription(_textureLayout, textureView));
-                rsi = new ResourceSetInfo(GetNextImGuiBindingID(), resourceSet);
+            var id = GetNextImGuiBindingID();
 
-                _setsByView.Add(textureView, rsi);
-                _viewsById.Add(rsi.ImGuiBinding, rsi);
-                _ownedResources.Add(resourceSet);
-            }
+            Textures.Add(new() {ID = id, Texture = rs});

```

Comment:
Maybe keeping track of resources using dictionaries that store keys to each other this way was a good idea, only having rs' stored as a list of Bindings seems to be failing
I think the best course of action is to remove the `List<Binding>` and have 3 dicts as follows:
- ID -> Texture
- Texture -> TextureView
- TextureView -> ResourceSet

Reply:
Turns out, 2 ID-indexed dicts pointing at the TextureView and ResourceSet are enough, Texture can be pulled from TV and the resourceset can easily be kept up-to-date with the TV/T